### ISSUE TITLE
Unable to create cloud private nat on gcp using google_compute_router_nat

### ISSUE THREAD

**Issue by Pthuturkar** [terraform-google-modules/terraform-google-cloud-nat#169]
### TL;DR

Description: type parameter is not working in the resource google_compute_router_nat

Reference : https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_router_nat#example-usage---router-nat-private

### Expected behavior

Resource should be able to create private cloud nat  on gcp 

### Observed behavior

error :  Error: Unsupported argument
│ 
│   on .terraform/modules/cloud-nat-test/cloud_nat.tf line 20, in resource "google_compute_router_nat" "cloud_nat":
│   20:   type                                = "PRIVATE"
│ 
│ An argument named "type" is not expected here.

### Terraform Configuration

```hcl
resource "google_compute_router_nat" "cloud_nat" {
  project                             = var.project_id
  name                                = var.cloud_nat_name
  router                              = google_compute_router.router.name
  region                              = google_compute_router.router.region
  source_subnetwork_ip_ranges_to_nat  = var.source_subnetwork_ip_ranges_to_nat
  enable_endpoint_independent_mapping = var.enable_endpoint_independent_mapping
  min_ports_per_vm                    = var.min_ports_per_vm
  nat_ip_allocate_option              = var.nat_ip_allocate_option
  type                                = "PRIVATE"
  subnetwork {
    name                    = google_compute_subnetwork.subnet.id
    source_ip_ranges_to_nat = ["ALL_IP_RANGES"]
  }
  rules {
    rule_number = var.rule_number
    description = "rule for private nat"
    match       = var.match
    action {
      source_nat_active_ranges = [
        data.google_compute_subnetwork.vpc_subnetwork.id
      ]
    }
  }
}
```

### Terraform Version

```sh
terraform {
  required_providers {
    google = {
       source  = "hashicorp/google"
       version = "6.4.0"
    }
  }
}
```
